src/BlocklyInterface/BlocklyInstance.ts

```diff
@@ -0,0 +1,47 @@
+import Blockly from "blockly";
+import "blockly/javascript";
+
+declare interface BlocklyJavaScript {
+  STATEMENT_PREFIX: string;
+  addReservedWords: any;
+  workspaceToCode: any;
+}
+
+const BLOCKLY_HIGHLIGHT_PREFIX = "highlightBlock";
+
+// Class wrapping blockly providing methods to directly access it. React/redux
+// interaction is handled above this.
+class BlocklyInstance {
+  private workspace: Blockly.WorkspaceSvg;
+
+  constructor(workspaceArea: HTMLDivElement, toolbox: HTMLElement) {
+    this.workspace = Blockly.inject(workspaceArea, { toolbox });
+
+    this.setupInterpretation();
+    this.resizeBlockly();
+  }
+
+  setupInterpretation() {
+    const generator = (Blockly as any).JavaScript as BlocklyJavaScript;
```

Comment:
This is of course the wrong way to do this, but looks like the way Blockly generators are imported is broken with their typescript definitions (i.e. importing of the specified generator after Blockly is imported doesn't change the type definitions). I tried merging in declarations but got nowhere.

Anyone have any good ideas?

Reply:
Is the issue that Blockly is missing types so you are trying to type Blockly? If so, you can make a .d.ts file that defines blockly and its types.

Reply:
It's slightly more tricky than that. Blocky defines a .d.ts, but it doesn't include it for the generators (as they aren't necessarily all used). In the second blocky import at the top of the file, the import adds the JavaScript key to the Blocky module, but doesn't do any types.

Reply:
Can we have a single function that returns the typed generator?